src/components/model-authoring/edit-name.tsx

```diff
@@ -0,0 +1,38 @@
+import * as React from "react";
+import { useState } from "react";
+
+import * as css from "./edit-name.scss";
+
+interface IProps {
+  name: string;
+  saveName: (name: string) => void;
+}
+
+export const EditName = ({ name, saveName }: IProps) => {
+  const [newName, setNewName] = useState<string>(name);
+  const [disabled, setDisabled] = useState<boolean>(true);
```

Comment:
I think this might be better named `[editing, setEditing]` with a default value of `false`